### PR TITLE
Network swarm: Split functions and reduce indentation.

### PR DESCRIPTION
**Pull request by nibhar**
This PR does not change any functionality. It only restructures the `handle_event` function and some other minor locations in that file. I was doing these on the fly as I went and the changes would have been very hard to track, so I decided to bundle them up and make a separate PR for them to ease reviewing.

### CODE REVIEW

**Comment by hrxi** [comment]
[Diff without whitespace](https://github.com/nimiq/core-rs-albatross/pull/3012/files?w=1).

File: network-libp2p/src/swarm.rs [hunk 1]
```diff
@@ -408,517 +411,579 @@ fn handle_event(
...
+                    .unwrap()
+                    .finish();
+            }
+        }
+        Ok(GetRecordOk::FinishedWithNoAdditionalRecord { cache_candidates }) => {
+            // Remove the query, send the best result to the application layer
+            // and push the best result to the cache candidates
+
+            let Some(results) = event_info.state.dht_get_results.remove(&id) else {
+                panic!("DHT inconsistent state, query_id: {:?}", id);
```
Comment (Eligioo) [inline network-libp2p/src/swarm.rs:567]: Do we need to panic here?

**Review by Eligioo — COMMENTED** [review]
Nice effort! Added one question.

Decoupling these events from its handlers also already prepares how we can eventually split our current `Behaviour` and introduce a 'light' `Behaviour` 👍

File: network-libp2p/src/swarm.rs (hunk 1 above)
Comment (nibhar) [reply network-libp2p/src/swarm.rs:567]: The panic was there before. I was considering to remove it, but choose not to do it in this pr, even if it should change, as it is supposed to not alter how the code works.

**Review by Eligioo — APPROVED** [review]

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**